Support Shadowserver Vulnerable SMTP server feed, fix #1984

monoidic opened the pull request:
This PR adds support for the Shadowserver Vulnerable SMTP Report and fixes #1984 by differentiating between Vulnerable-HTTP and Accessible-HTTP filenames.

wagner-certat commented on intelmq/bots/parsers/shadowserver/_config.py:
```diff
@@ -2864,6 +2890,7 @@ def scan_exchange_identifier(field):
     ('Vulnerable-ISAKMP', 'scan_isakmp', vulnerable_isakmp),
     ('Vulnerable-HTTP', 'scan_http', accessible_vulnerable_http),
     ('Vulnerable-Exchange-Server', 'scan_exchange', scan_exchange),
+    ('Vulnerable-SMTP', 'scan_smtp', vulnerable_smtp),
```
```suggestion
    ('Vulnerable-SMTP', 'scan_smtp_vulnerable', vulnerable_smtp),
```
In our case the filename is ` 2021-08-12-scan_smtp_vulnerable-austria-geo.csv.zip`

wagner-certat reviewed (changes requested):
Could you please add an entry in `docs/user/bots.rst` in the shadowserver feed section? Thanks

otherwise ready for merge

wagner-certat commented on intelmq/bots/parsers/shadowserver/_config.py:
```diff
@@ -2862,8 +2888,9 @@ def scan_exchange_identifier(field):
     ('Spam-URL', 'spam_url', spam_url),
     ('Vulnerable-ISAKMP', 'scan_isakmp', vulnerable_isakmp),
-    ('Vulnerable-HTTP', 'scan_http', accessible_vulnerable_http),
+    ('Vulnerable-HTTP', 'scan_http_vulnerable', accessible_vulnerable_http),
```
Good catch, thanks!

monoidic commented:
The suggested improvement reminded me of #1984, so I also fixed that by making Vulnerable-HTTP match by `scan_http_vulnerable` rather than `scan_http`, and added some standalone tests for `scan_http_vulnerable`.

wagner-certat commented:
> The suggested improvement reminded me of #1984,

yeah, me too - but only after you committed the other fix :)

> so I also fixed that by making Vulnerable-HTTP match by `scan_http_vulnerable` rather than `scan_http`, and added some standalone tests for Vulnerable-HTTP (previously, only Accessible-HTTP had tests).

Thanks! Didn't think that #1984 is based on such a simple mistake.

wagner-certat commented:
Thanks! Also cherry-picked for maintenance in 7a81e3709 plus changelog in b76520eac
